source/MRTest/MRPointCloudVariadicOffsetTests.cpp

```diff
@@ -86,13 +90,29 @@ TEST( MRMesh, findClosestWeightedMeshPoint )
         params.pointWeight = [&]( VertId ) { return 1; };
         params.maxWeight = 1;
         for ( float z = -2; z <= 2; z += 0.1f )
-            EXPECT_NEAR( distance( Vector3f( 0, 0, z ) ), -1 + std::abs( z ), 1e-7f );
+            EXPECT_NEAR( distance( Vector3f( 0, 0, z ), params.bidirectionalMode ), -1 + std::abs( z ), 1e-7f );
 
         params.pointWeight = [&]( VertId ) { return -1; };
         params.maxWeight = -1;
         for ( float z = -2; z <= 2; z += 0.1f )
-            EXPECT_NEAR( distance( Vector3f( 0, 0, z ) ),  1 + std::abs( z ), 1e-7f );
+            EXPECT_NEAR( distance( Vector3f( 0, 0, z ), params.bidirectionalMode ),  1 + std::abs( z ), 1e-7f );
     }
 }
 
+TEST( MRMesh, weightedMeshShell )
+{
+    auto cube = makeCube();
+    auto rot = Matrix3f::rotation( { 1.f, 1.f, 1.f }, PI_F / 6.f );
+    for ( auto& pt : cube.points )
+        pt = rot * pt;
+
+    ASSERT_TRUE( remesh( cube, { .maxEdgeSplits = 10000 } ) );
```

Comment:
Without subdivision the test will work faster (which is good), but will it loose something?